IzvorniKod/frontend/src/components/PostPreview/PostPreview.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { HeartIcon, HandThumbUpIcon, HandThumbDownIcon, MegaphoneIcon, TrashIcon } from "@heroicons/react/24/solid";
+
+ type PostPreviewType = {
+    postTitle: string;
+    onClick: () => void;
+  };
+  
+  export const PostPreview = ({
+    postTitle,
+    onClick,
+  }: PostPreviewType) => {
+    
+    const postPreviewDiv = useRef<HTMLDivElement>(null);
+    const [size, setSize] = useState<{ width: number; height: number }>({ width: 0, height: 0 });
+
+    useEffect(() => {
+        if (postPreviewDiv.current) {
+          const resizeObserver = new ResizeObserver(() => {
+            if (postPreviewDiv.current) {
+              const { width } = postPreviewDiv.current.getBoundingClientRect();
+              setSize({ width, height: 0 });
+            }
+          });
+    
+          resizeObserver.observe(postPreviewDiv.current); 
+          return () => resizeObserver.disconnect();
+        }
+      }, []);
+
+    const getFormattedTitle = () => {
+        const charWidth = 18; 
+        const maxChars = Math.floor(size.width / charWidth); 
+        return postTitle.length > maxChars ? `${postTitle.slice(0, maxChars)}...` : postTitle;
+      };
+
+    const [isHeartClicked, setIsHeartClicked] = useState(false);
+    const [isThumbUpClicked, setIsThumbUpClicked] = useState(false);
+    const [isThumbDownClicked, setIsThumbDownClicked] = useState(false);
+    const [isReportClicked, setIsReportClicked] = useState(false);
+    const [isTrashClicked, setIsTrashClicked] = useState(false);
+
+    const getHearthStyle = (isClicked: boolean) => {
+        return {
+        stroke: isClicked ? 'none' : 'black',
+        strokeWidth: isClicked ? '0' : '1px',
+        fill: isClicked ? 'red' : 'white', 
+        };
+    };
+    const getThumbsStyle = (isClicked: boolean) => {
+        return {
+        stroke: isClicked ? 'none' : 'black',
+        strokeWidth: isClicked ? '0' : '1px',
+        fill: isClicked ? 'blue' : 'white', 
+        };
+    };
+    const getBlackStyle = (isClicked: boolean) => {
+        return {
+        stroke: isClicked ? 'none' : 'black',
+        strokeWidth: isClicked ? '0' : '1px',
+        fill: isClicked ? 'black' : 'white', 
+        };
+    };
```

Comment:
od ovoga napravi jednu getIconStyle funkciju koja kao parametre prima isClicked: boolean; i color: string; u obliku tipiziranog objekta. Tip treba izgledati ovako:
`type GetIconStyleType = {`
`     isClicked: boolean;`
`    color: string;`
`}`

i onda ga koristis u funciji kao:

`const getIconStyle = ({color, isClicked}: GetIconStyleType) ....`

nakon toga mozes umjesto svih onClick funkcija na svaku ikonicu postavit ovu sa trazenom bojom

bilo bi pozeljno to napravit sa klasama ali za prvu ruku je okej i kroz style (ako ces radit sa klasama i nesto ti nece bit jasno pitaj pa pomognem)

Reply:
5048ba2d064d7d2b864da6164ccbf74308fc1cc1